src/ngx_http_lua_shdict.c

```diff
@@ -2983,6 +2983,24 @@ ngx_http_lua_ffi_shdict_set_expire(ngx_shm_zone_t *zone, u_char *key,
 
     return NGX_OK;
 }
+
+
+#if nginx_version >= 1011007
+void
+ngx_http_lua_ffi_shdict_get_stats(ngx_shm_zone_t *zone,
+    size_t *free_page_bytes)
+{
+    ngx_http_lua_shdict_ctx_t   *ctx;
+
+    ctx = zone->data;
+
+    ngx_shmtx_lock(&ctx->shpool->mutex);
+
+    *free_page_bytes = ctx->shpool->pfree * ngx_pagesize;
+
+    ngx_shmtx_unlock(&ctx->shpool->mutex);
+}
+#endif /* nginx_version >= 1011007 */
 #endif /* NGX_LUA_NO_FFI_API */
```

Comment:
It would be nicer to add a blank line before these 2 `#endif` lines.

Reply:
done